rust/sdk-processor/src/steps/ans_processor/ans_extractor.rs

```diff
@@ -0,0 +1,151 @@
+        ans_lookup::{AnsLookup, AnsPrimaryName, CurrentAnsLookup, CurrentAnsPrimaryName},
+        ans_lookup_v2::{
+            AnsLookupV2, AnsPrimaryNameV2, CurrentAnsLookupV2, CurrentAnsPrimaryNameV2,
+        },
+    },
+    processors::ans_processor::parse_ans,
+    worker::TableFlags,
+};
+
+pub struct AnsExtractor
+where
+    Self: Sized + Send + 'static,
+{
+    deprecated_table_flags: TableFlags,
+    config: AnsProcessorConfig,
+}
+
+impl AnsExtractor {
+    pub fn new(
+        deprecated_table_flags: TableFlags,
+        config: ProcessorConfig,
+    ) -> Result<Self, anyhow::Error> {
+        let processor_config = match config {
+            ProcessorConfig::AnsProcessor(processor_config) => processor_config,
+            _ => {
+                return Err(anyhow::anyhow!(
+                    "Invalid processor config for ANS Processor: {:?}",
+                    config
+                ))
+            },
+        };
+
+        Ok(Self {
+            deprecated_table_flags,
+            config: processor_config,
+        })
+    }
+}
+
+#[async_trait]
+impl Processable for AnsExtractor {
+    type Input = Vec<Transaction>;
+    type Output = (
+        Vec<CurrentAnsLookup>,
+        Vec<AnsLookup>,
+        Vec<CurrentAnsPrimaryName>,
+        Vec<AnsPrimaryName>,
+        Vec<CurrentAnsLookupV2>,
+        Vec<AnsLookupV2>,
+        Vec<CurrentAnsPrimaryNameV2>,
+        Vec<AnsPrimaryNameV2>,
+    );
+    type RunType = AsyncRunType;
+
+    async fn process(
+        &mut self,
+        input: TransactionContext<Vec<Transaction>>,
+    ) -> Result<
+        Option<
+            TransactionContext<(
+                Vec<CurrentAnsLookup>,
+                Vec<AnsLookup>,
+                Vec<CurrentAnsPrimaryName>,
+                Vec<AnsPrimaryName>,
+                Vec<CurrentAnsLookupV2>,
+                Vec<AnsLookupV2>,
+                Vec<CurrentAnsPrimaryNameV2>,
+                Vec<AnsPrimaryNameV2>,
+            )>,
+        >,
+        ProcessorError,
+    > {
+        let (
+            mut all_current_ans_lookups,
+            mut all_ans_lookups,
+            mut all_current_ans_primary_names,
+            mut all_ans_primary_names,
+            all_current_ans_lookups_v2,
+            all_ans_lookups_v2,
+            all_current_ans_primary_names_v2,
+            mut all_ans_primary_names_v2,
+        ) = parse_ans(
+            &input.data,
+            self.config.ans_v1_primary_names_table_handle.clone(),
+            self.config.ans_v1_name_records_table_handle.clone(),
+            self.config.ans_v2_contract_address.clone(),
+        );
+
+        if self
+            .deprecated_table_flags
+            .contains(TableFlags::ANS_PRIMARY_NAME)
+        {
+            all_ans_primary_names.clear();
+        }
+        if self
+            .deprecated_table_flags
+            .contains(TableFlags::ANS_PRIMARY_NAME_V2)
+        {
+            all_ans_primary_names_v2.clear();
+        }
```

Comment:
this is deprecated, we can remove parsing `ANS_PRIMARY_NAME_V2`

Reply:
thanks, removed